File: src/repository/OneWayDao.java
package repository;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;
import java.util.function.Consumer;

import javax.persistence.EntityManager;
import javax.persistence.EntityManagerFactory;
import javax.persistence.EntityTransaction;
import javax.persistence.Persistence;
import javax.persistence.Query;

import model.OneWay;
import model.User;

public class OneWayDao implements Repository<OneWay, Integer>{
	
	private EntityManagerFactory emf = Persistence.createEntityManagerFactory(
			"jdbc:sqlserver://localhost/Alibaba.odb;integratedSecurity=true;");
	private EntityManager em=emf.createEntityManager(); 

	@Override
	public OneWay findbyId(Integer id) {
		try {
			return em.find(OneWay.class, id);
		} catch (Exception e) {
			return null;
		}
	}

	@Override
	public List<OneWay> findByIDs(Collection<Integer> ids) {
		List<OneWay> users = new ArrayList<OneWay>();
		try {
			for(int id : ids) {
				users.add(em.find(OneWay.class, id));
			}
			return users;
		} catch (Exception e) {
			return null;
		}
	}

	@Override
	public List<OneWay> findAll() {
		Query q = em.createQuery("select * from One_Way");
		return q.getResultList();
	}

	@Override
	public Boolean deleteByID(Integer id) {
		try {
			executeInsideTransaction(entityManager -> em.remove(em.find(OneWay.class, id)));
			return true;
		} catch(Exception e) {
			return false;
		}
	}

	@Override
	public Boolean DeleteByIDs(Collection<Integer> ids) {
		try {
			for(int id : ids) {
				executeInsideTransaction(entityManager -> em.remove(em.find(OneWay.class, id)));
			}
			return true;
		} catch(Exception e) {
			return false;
		}
	}

	@Override
	public OneWay save(OneWay E) {
		try {
			executeInsideTransaction(entityManager -> entityManager.persist(E));
			return em.find(OneWay.class, emf.getPersistenceUnitUtil().getIdentifier(E));
		} catch(Exception e) {
			executeInsideTransaction(entityManager -> entityManager.merge(E));
			return em.find(OneWay.class, emf.getPersistenceUnitUtil().getIdentifier(E));
		}
	}
	
	private void executeInsideTransaction(Consumer<EntityManager> action) {
        EntityTransaction tx = em.getTransaction();
        try {
            tx.begin();
            action.accept(em);
            tx.commit(); 
        }
        catch (RuntimeException e) {
            tx.rollback();
            throw e;
        }
    }

}

File: repository/OneWayDao.py
from Repository import RepositoryInterface
from Connection import getConn
from OneWay import OneWay
class OneWayDao(RepositoryInterface):
    def findById(id):
        try:
            c = getConn()
            c.execute('select * from One_Way where DomesticFlightTransportnumber=' + str(id))
            i = c.fetchall()
            return OneWay(i[0][0], i[0][1])
        except:
            print("There was an error")
            return None
    def findByIds(ids):
        try:
            a = []
            c = getConn()
            for id in ids:
                c.execute('select * from One_Way where DomesticFlightTransportnumber=' + str(id))
                i = c.fetchall()
                a.append(OneWay(i[0][0], i[0][1]))
            return a
        except:
            print("There was an error")
            return None
    def findAll():
        try:
            a = []
            c=getConn()
            c.execute('select * from One_Way')    
            for i in c:
                a.append(OneWay(i[0], i[1]))
            return a
        except:
            print("There was an error")
            return None
    def deleteById(id):
        try:
            c=getConn()
            c.execute('delete from One_Way where DomesticFlightTransportnumber=' + str(id))
            c.commit()
            return True
        except:
            return False
    def deleteByIds(ids):
        try:
            c=getConn()
            for id in ids:
                   c.execute('delete from One_Way where DomesticFlightTransportnumber=' + str(id))
                   c.commit()
            return True
        except:
            return False
    def save(entity):
        try:
            c=getConn()
            c.execute('insert into One_Way values(' + str(entity.DomesticFlightTransportnumber)+ str(entity.OverseasFlightTransportnumber)+')')
            c.commit()
            return OneWay(entity.FlightTransportnumber, entity.OverseasFlightTransportnumber)
        except:
            print("There was an error")
            return None